RotationSolver/Updaters/TargetUpdater.cs

```diff
@@ -521,4 +524,28 @@ private static void UpdateNamePlate(IEnumerable<BattleChara> allTargets)
         }
         DataCenter.TreasureCharas = charas.ToArray();
     }
+
+    // status checks that can't easily be represented in config
+    private static bool CheckTargetableOther(BattleChara b) {
+        // Labyrinth of the Ancients
+        if (Svc.ClientState.TerritoryType == 174 && b.Name.ToString() == "Thanatos") {
+            // thanatos can only be damaged by players with Astral Realignment (398)
+            return Player.Status.Any(s => s.StatusId == 398);
+            // TODO: Allagan Bomb is also invulnerable until everything else is dead, but that's harder to check
+        }
+
+        // The Puppets' Bunker
+        if (Svc.ClientState.TerritoryType == 917) {
+            // second boss has one of three buffs corresponding to an alliance letter
+            // player needs the matching buff
+            var minibossBuff = b.StatusList.Where(x => x.StatusId == 2409 || x.StatusId == 2410 || x.StatusId == 2411).FirstOrDefault();
+            // 2288 -> 2409
+            // 2289 -> 2410
+            // 2290 -> 2411
+            if (minibossBuff != null)
+                return Player.Status.Any(s => s.StatusId == minibossBuff.StatusId - 121);
```

Comment:
Please put this into a variable just after the if statement, then use it, otherwise this would be calculated for each status the Player has until the condition is met, or otherwise you've exhausted the list of statuses

Reply:
Sorry, I'm not following, what should be put into a variable here?

Reply:
Ah, sorry, my bad! `minibossBuff.StatusId - 121` part. That would be calculated for each status until you find one that is true